src/pages/DistributePage/DistributePage.tsx

```diff
@@ -0,0 +1,239 @@
+import { useState, ChangeEvent } from 'react';
+
+import { useParams } from 'react-router-dom';
+
+import {
+  FormControl,
+  InputLabel,
+  MenuItem,
+  Select,
+  Button,
+} from '@material-ui/core';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+
+import { Link, Box, Panel } from '../../ui';
+import { FormTextField } from 'components';
+import { useCurrentOrg } from 'hooks/gql';
+import { useSelectedCircle } from 'recoilState';
+import { useVaults } from 'recoilState/vaults';
+import * as paths from 'routes/paths';
+
+import AllocationTable from './AllocationsTable';
+
+/**
+ * Displays a list of allocations and allows generation of Merkle Root for a given circle and epoch.
+ * @param circleId string
+ * @param epochId string
+ * @returns JSX.Element
```

Comment:
So the only reason I include this type of commenting is to help with Intellisense on VS Code. Adding these comments is part of [clean code](https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29) philosophy and it helps onboard new developers when they can just simply hover a component implementation and see details about the intent. Now, this can also be assisted when you have proper variable naming but I think for components that are reused in other files in the codebase, a general good rule is to ensure that there's something that describes intent.